From 006c3fee908b797061c602878584ea5908a5eadf Mon Sep 17 00:00:00 2001
From: Ramazan Yestekbayev <ramazan.yestekbayev@sintecs.eu>
Date: Wed, 25 Mar 2026 15:27:24 +0200
Subject: [PATCH] plat-ls: Add LS1028A dynamic SHM memory registration

Signed-off-by: ramazan-yestekbayev <ramazan.yestekbayev@sintecs.eu>
---
 core/arch/arm/plat-ls/main.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/core/arch/arm/plat-ls/main.c b/core/arch/arm/plat-ls/main.c
index 7e013f9c2..c7a1cc266 100644
--- a/core/arch/arm/plat-ls/main.c
+++ b/core/arch/arm/plat-ls/main.c
@@ -62,6 +62,13 @@ register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
 register_phys_mem_pgdir(MEM_AREA_IO_SEC, GIC_BASE, CORE_MMU_PGDIR_SIZE);
 #endif
 
+#if defined(PLATFORM_FLAVOR_ls1028ardb)
+	register_ddr(CFG_DRAM0_BASE, (CFG_TZDRAM_START - CFG_DRAM0_BASE));
+	#ifdef CFG_DRAM1_BASE
+		register_ddr(CFG_DRAM1_BASE, CFG_DRAM1_SIZE);
+	#endif
+#endif
+
 #if defined(PLATFORM_FLAVOR_lx2160ardb) || defined(PLATFORM_FLAVOR_lx2160aqds)
 register_ddr(CFG_DRAM0_BASE, (CFG_TZDRAM_START - CFG_DRAM0_BASE));
 #ifdef CFG_DRAM1_BASE
-- 
2.43.0

